crate/utils/src/crypto/rsa.rs

```diff
@@ -0,0 +1,131 @@
+use cosmian_kmip::kmip::{
+    kmip_data_structures::{KeyBlock, KeyMaterial, KeyValue},
+    kmip_objects::{Object, ObjectType},
+    kmip_types::{
+        Attributes, CryptographicAlgorithm, CryptographicDomainParameters, CryptographicParameters,
+        CryptographicUsageMask, KeyFormatType, Link, LinkType, LinkedObjectIdentifier,
+    },
+};
+use num_bigint_dig::BigUint;
+use openssl::{pkey::Private, rsa::Rsa};
+use tracing::trace;
+
+/// convert to RSA KMIP Public Key
+pub fn to_rsa_public_key(
+    private_key: &Rsa<Private>,
+    pkey_bits_number: u32,
+    public_key_uid: &str,
+) -> Object {
+    let cryptographic_length_in_bits = private_key.e().num_bits();
```

Comment:
Why would you take the public exponent as length ? better take the modulus
```suggestion
    let cryptographic_length_in_bits = private_key.n().num_bits();
```

Reply:
thanks